src/qt/pivx/txviewholder.cpp

```diff
@@ -35,11 +35,23 @@ void TxViewHolder::init(QWidget* holder, const QModelIndex &index, bool isHovere
             type !=  TransactionRecord::ZerocoinSpend_Change_zPiv &&
             type !=  TransactionRecord::StakeZPIV &&
             type != TransactionRecord::Other) {
+
         QString address = rIndex.data(Qt::DisplayRole).toString();
-        if (address.length() > 20) {
-            address = address.left(ADDRESS_SIZE) + "..." + address.right(ADDRESS_SIZE);
+        if (!address.isEmpty()) {
+            if (type == TransactionRecord::SendToNobody) {
+                // OP_RETURN record, check if it's from a valid utf-8 string to show
+                if (IsValidUTF8(address.toStdString())) {
```

Comment:
no need add this validation, as `QString::toStdString()` already returns valid utf-8 (https://doc.qt.io/qt-5/qstring.html#toStdString)

Reply:
k, will move it to the record parsing side.